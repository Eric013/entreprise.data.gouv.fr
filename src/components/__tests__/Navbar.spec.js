import { createLocalVue, shallow } from '@vue/test-utils'
import Vuex from 'vuex'
import  NavBar from '@/components/Navbar.vue'
import { __createMocks as createStoreMocks } from '@/store/index.js';

const localVue = createLocalVue()
localVue.use(Vuex)
jest.mock('@/store/index.js')

describe('Navbar.vue', () => {
  let storeMocks
  let wrapperNavBar

  // Create a fresh store and wrapper instance for every test case.
  beforeEach(() => {
    storeMocks = createStoreMocks()

    wrapperNavBar = shallow(NavBar, {
      localVue,
      store: storeMocks.store
    })
  })

  test('It calls goToClearedHomePage when clicking on the logo', () => {
    wrapperNavBar.find('.nav__link').trigger('click')
    expect(storeMocks.actions.goToClearedHomePage.mock.calls).toHaveLength(1)
  })
})
