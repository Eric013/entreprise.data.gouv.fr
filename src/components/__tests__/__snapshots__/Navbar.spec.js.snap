// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results.vue snapshot testing It match the snapshot 1`] = `
<nav
  class="nav"
>
  <div
    class="nav__container"
  >
    <a
      class="nav__link"
    >
      <img
        alt="entreprise.data.gouv.fr"
        class="nav__logo"
        src="../assets/img/entreprise.svg"
      />
    </a>
  </div>
</nav>
`;
